vumi_http_proxy/servicemaker.py

```diff
@@ -27,7 +29,12 @@ class ProxyWorkerServiceMaker(object):
     options = Options
 
     def makeService(self, options):
+        try:
+            port = int(options["port"])
+        except ValueError:
+            log.err('Port must be an integer. Please try again')
+            sys.exit()
```

Comment:
This check should probably happen in https://twistedmatrix.com/documents/15.5.0/api/twisted.python.usage.Options.html#postOptions (just add a `postOptions` method to your existing options). You can then raise `UsageError("nice message")` instead of calling `sys.exit`.